### PR TITLE
Update: Allow for unselectable tree nodes

### PR DESCRIPTION
**Pull request by etaoins**
For segment targeting sometimes we need e.g. country nodes to show a path to specific regions but the country itself isn't selectable. To support this add an optional `isSelectable` property to the tree nodes which defaults to true.

### CODE REVIEW

File: test/mocks/TreePickerMocks.js [hunk 1]
```diff
@@ -22,8 +22,14 @@ const rootTypes = [
 const auPath = { id: 'au', label: 'AU' };
 const actPath = { id: 'au-act', label: 'ACT' };
 
-const actNode =
-  { id: 'au-act', label: 'Australian Capital Territory', type: 'State', path: [auPath], value: 1000, rootTypeId: 'a' };
+const actNode = {
+  id: 'au-act',
+  label: 'Australian Capital Territory',
+  type: 'State',
+  path: [auPath], value: 1000, rootTypeId: 'a',
```
Comment (omgaz) [inline test/mocks/TreePickerMocks.js]: Can these three be on their own lines?

**Comment by omgaz** [comment]
`Gary requesting 2 change(s)`

**Comment by etaoins** [comment]
@omgaz Didn't even notice that. Looking good now?

**Comment by omgaz** [comment]
`omgaz approving 1 change(s)`

**Comment by camjc** [comment]
@etaoins Ya'll don't want to assert that a node is at least either *selectable* or *expandable*?
Shouldn't slow down perf much and would stop consumers putting their nodes into an unusable state.

**Pull request merged**